File: 121_Best_Time_To_Buy_Sell_Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxprofit = 0;
        int minvalue = INT_MAX;
        for(int i = 0; i<prices.size(); i++){
            //if the current price is lower than current lowest value
            minvalue = min(prices[i],minvalue);
            
            //profit = what was previously calculated or new value if lower
            maxprofit= max(maxprofit,prices[i] - minvalue);
        }
        return maxprofit;
    }
};

File: 121_Best_Time_To_Buy_Sell_stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        highest_profit = 0
        smallest_value = 100000000 #initialized to a large value to be able to be compared in loop
        for price in prices:
            smallest_value = min(smallest_value, price)
            highest_profit = max(highest_profit, price - smallest_value)
        return highest_profit
